kong/plugins/zipkin/reporter.lua

```diff
@@ -21,17 +23,35 @@ local function ip_kind(addr)
 end
 
 
-local function new(http_endpoint, default_service_name, local_service_name, connect_timeout, send_timeout, read_timeout, logger)
+local function send_entries_to_zipkin(conf, entries)
+  if conf.http_endpoint == nil or conf.http_endpoint == ngx.null then
+    return true
+  end
+
+  kong.log.debug("zipkin batch size: ", #entries)
+  local httpc = resty_http.new()
+  httpc:set_timeouts(conf.connect_timeout, conf.send_timeout, conf.read_timeout)
+  local res, err = httpc:request_uri(conf.http_endpoint, {
+    method = "POST",
+    headers = {
+      ["content-type"] = "application/json",
+    },
+    body = cjson.encode(entries),
+  })
+  if not res then
+    return nil, "zipkin request failed: " .. err
+  elseif res.status < 200 or res.status >= 300 then
+    return nil, "zipkin server responded unexpectedly: " .. res.status .. " " .. res.reason
```

Comment:
I see that this is a 1:1 move from the existing code, but I'd safeguard `res.status` and `res.reason` access with either casting it `tostring()` or checking their presence before.